server/controllers/droppedAssets/handleQuestItemClicked.ts

```diff
@@ -24,27 +25,30 @@ export const handleQuestItemClicked = async (req: Request, res: Response) => {
 
     const analytics = [];
 
+    const visitor = await Visitor.create(visitorId, urlSlug, { credentials });
+
     const droppedAsset = await DroppedAsset.get(assetId, urlSlug, { credentials });
 
-    const { dataObject, world } = await getWorldDetails(credentials);
+    const { dataObject, world } = await getWorldDetails(credentials, true);
     const { itemsCollectedByUser, numberAllowedToCollect } = dataObject as DataObjectType;
 
     const lastCollectedDate = itemsCollectedByUser?.[profileId]?.lastCollectedDate;
     const differenceInDays = getDifferenceInDays(currentDate, new Date(lastCollectedDate));
-    const hasCollectedToday = differenceInDays === 0
+    const hasCollectedToday = differenceInDays === 0;
     let totalCollectedToday = 1,
       total = 1;
 
-    if (
-      hasCollectedToday &&
-      itemsCollectedByUser?.[profileId]?.totalCollectedToday >= numberAllowedToCollect
-    ) {
+    if (!hasCollectedToday) analytics.push({ analyticName: "starts", profileId, urlSlug, uniqueKey: profileId });
+
+    if (hasCollectedToday && itemsCollectedByUser?.[profileId]?.totalCollectedToday >= numberAllowedToCollect) {
       return res.json({ addedClick: false, numberAllowedToCollect, success: true });
     } else {
       const promises = [];
+      analytics.push({ analyticName: "itemsCollected" });
 
       // Move the quest item to a new random location
       const position = getRandomCoordinates(world.width, world.height);
+      promises.push(world.triggerParticle({ position: droppedAsset.position, name: "Smoke" }));
```

Comment:
Same with tic-tac-toe here...I think particle names should be envs with fallback default